fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2597,8 +2605,9 @@ private boolean areAllRecordTypesSince(@Nullable Collection<RecordType> recordTy
                     }
                     omitUnsplitRecordSuffix = formatVersion < SAVE_UNSPLIT_WITH_SUFFIX_FORMAT_VERSION;
                     info.clearOmitUnsplitRecordSuffix();
+                    addRecordsReadConflict(); // We used snapshot to determine emptiness, and are now acting on it.
```

Comment:
I don't think adding this read conflict range is wrong, so it's probably best to include it, but would other operations have failed anyway on conflicts reading the store-header key?

Reply:
That's indeed what prevents the problem in the other direction, where a transaction with an older idea of the format / meta-data / omit flag wants to commit after it's just been changed. That's generally true and not really specific to this case.

Reply:
Oh, right. But it's the case where this transaction reads the full store, someone else writes a new record at the old meta-data version, then this runs that is the problem.